Merge/more 3.0.x

landryb opened the pull request:
This remerges upstream/3.0.x into our branch, fixing some xsl formatter issues (geonetwork/core-geonetwork#1423), geonetwork/core-geonetwork#1448 looks a bit serious, contains 3 other small PRs from me, and on top of this i cherry-picked fc696fa53193d7433e95d00fa3b1fbb4056d599c which is needed for georchestra/georchestra#1268

landryb committed:
Only take the first IP address when parsing client IP.

When there are several (reverse-)proxies between the client and the
server, the header might contain several IP addresses, separated
by a comma. This was leading to NumberFormatException at each client
request, trying to parse for example "109, 10"...

landryb committed:
Enforce dependency on httpclient 4.3.2 in toplevel pom.xml (#1368)

josegar74 committed:
Fix missing template getMetadataHeader, required in xsl formatter

josegar74 committed:
XSL formatter broken #1423

landryb committed:
Add a redirect rule for ExtJS-style permalinks (#808)

Redirects /apps/search/?uuid=xx to /#/metadata/xx

fxprunayre committed:
Inspire theme Population distribution — demography not shown #1426.

fxprunayre committed:
Merge remote-tracking branch 'origin/3.0.x' into 3.0.x

Commit:
Merge pull request #1425 from landryb/fix/refix-808

Add a redirect rule for ExtJS-style permalinks (#808)

Commit:
Merge pull request #1424 from landryb/fix/1179-3.x

Only take the first IP address when parsing client IP.

Commit:
Merge pull request #1422 from landryb/fix/1368-again

Enforce dependency on httpclient 4.3.2 in toplevel pom.xml (#1368)

Delawen committed:
Do not get all the metadata table for UUIDMapper. If the harvester collects from a big catalog, it throws a memory error

Commit:
optimise translations, add missing i18n.xml

juanluisrp committed:
Fix the reference to the non-existent iso-....xsl

When the CWS files were renamed, the own-brief.xsl, own-full.xsl and
own-summary.xsl kept the old include of iso-XXXXX.xsl files. This update
that reference to gmd-XXXXXX.xsl files.

josegar74 committed:
Merge pull request #1447 from juanluisrp/fix-is19139-csw-own-xsl

Fix the reference to the non-existent iso-....xsl

ticheler committed:
Dockerfile for GeoNetwork

josegar74 committed:
#1448 Saving a metadata with an empty keyword value creates invalid metadata

Delawen committed:
Less stress on indexing when harvesting. Useful for large harvesters.

Commit:
Merge pull request #1444 from pvgenuchten/dutch-lang-fix2

optimise translations, add missing i18n.xml

josegar74 committed:
#1448 Add gmx namespace in schemaLocation for iso19139

Delawen committed:
Fixing d63a824348d4890da5e03ac62f77ac069c7c3d72 Missing flush

landryb committed:
Add runtime dependency on apache-log4j-extras (georchestra/georchestra#1268)

landryb committed:
Bump log4j requirement to 1.2.17.

1.2.14 ships with rollingPolicy in log4j.dtd
1.2.16 allows to configure rollingPolicy via properties files

landryb committed:
Merge remote-tracking branch 'upstream/3.0.x' into merge/more-3.0.x

pmauduit committed:
Merge pull request #1451 from landryb/fix/bump-log4j

Bump log4j requirement to 1.2.17.

Tested at runtime in the geOrchestra composition, no issue encountered so far, merging.

landryb committed:
Merge remote-tracking branch 'upstream/3.0.x' into merge/more-3.0.x

landryb commented:
Reupdated after remerging log4j bump, i thought i merged this one and not apache-log4j-extras...

landryb commented:
Tested runtime without issues, just need to update the datadir to have the proper schemas from the webapp.

pmauduit commented on domain/src/main/java/org/fao/geonet/repository/MetadataRepositoryImpl.java:
```diff
@@ -134,5 +135,22 @@ public Metadata findOneOldestByChangeDate() {
+       Query query =  _entityManager.createQuery(
+               "select new org.fao.geonet.repository.SimpleMetadata("
+               + "id, uuid, dataInfo.changeDate, dataInfo.type_JPAWorkaround) "
+                + "from Metadata where harvestInfo.uuid = '" + id + "'");
```
this should work:

```
_entityManager.createQuery(
                "select new org.fao.geonet.repository.SimpleMetadata("
                 "id, uuid, dataInfo.changeDate, dataInfo.type_JPAWorkaround) "
                  "from Metadata where harvestInfo.uuid = :id").setParameter("id", id);

```

Obviously untested, and probably need to be reported upstream

pmauduit committed:
domain - fixing SQLi

Using parametrized queries from hibernate instead of building the SQL by
hand. See
https://docs.jboss.org/hibernate/orm/3.2/api/org/hibernate/Query.html

pmauduit commented:
LGTM, please merge !
